### PR TITLE
feat: add iceoryx

### PR DESCRIPTION
**Pull request by wep21**
- add [iceoryx](https://github.com/eclipse-iceoryx/iceoryx)

### CODE REVIEW

**Review by fmeum — CHANGES_REQUESTED** [review]
The lockfile isn't needed here (or in the archive).

**Comment by wep21** [comment]
@fmeum I'm sorry, I removed the lockfile.

File: modules/iceoryx/2.0.5/source.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "url": "https://github.com/eclipse-iceoryx/iceoryx/archive/refs/tags/v2.0.5.tar.gz",
```
Comment (fmeum) [inline modules/iceoryx/2.0.5/source.json:2]: Does this repo offer release archives? Those are generally preferred as their hashes are guaranteed to remain stable over time, but it's also not a problem if they aren't available.

**Review by fmeum — APPROVED** [review]

**Comment by fmeum** [comment]
Looks like CI is failing due to missing build files, but the BCR setup looks good.

File: modules/iceoryx/2.0.5/source.json (hunk 1 above)
Comment (wep21) [reply modules/iceoryx/2.0.5/source.json:2]: Unfortunately, it doesn't offer release archives.

**Comment by wep21** [comment]
@fmeum Do you know how to handle patch adding empty file? It causes a patch error as below.
https://buildkite.com/bazel/bcr-presubmit/builds/2987#018ce4a0-352b-47bb-8c25-a5759631964c/236-266

**Comment by fmeum** [comment]
@wep21 How did you generate the patch? The Bazel implementation of patch can be a bit strict at times. I would recommend using `git diff` to generate the patch, those usually work.

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (iceoryx) have been updated in this PR. Please review the changes.

**Comment by wep21** [comment]
@fmeum @meteorcloudy ready to review

File: modules/iceoryx/2.95.3/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    version = "2.95.3",
+    bazel_compatibility = [">=7.2.1"],
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "cpptoml", version = "0.1.1")
+bazel_dep(name = "platforms", version = "0.0.10")
+bazel_dep(name = "rules_cc", version = "0.0.9")
+bazel_dep(
+    name = "buildifier_prebuilt",
```
Comment (fmeum) [inline modules/iceoryx/2.95.3/MODULE.bazel:12]: Can this be a dev dep? Same for googletest.

Comment (lalten) [reply modules/iceoryx/2.95.3/MODULE.bazel:12]: No: https://github.com/eclipse-iceoryx/iceoryx/pull/2371

**Review by fmeum — APPROVED** [review]

**Pull request merged**